src/main/java/org/skillset/mystays/repository/AccountService.java

```diff
@@ -39,13 +39,13 @@ public boolean insert(Account account) {
             ac.lastName = account.getLastName();
             ac.password = account.getPassword();
             ac.phone = account.getPhoneNumber();
-            ac.persist();
+            ac.persistAndFlush();
 
             UserAddress userAddress= new UserAddress();
             userAddress.address = account.getFirstName() + " address";
             userAddress.user = ac;
             userAddress.zipcode = "23456";
-            userAddress.persist();
+            userAddress.persistAndFlush();
             return true;
         } catch (Exception e) {
```

Comment:
May be we can catch and handle specific exception for Account already exist case and return false for that case only. Here all the general exceptions, it will return false and will show Account already exist message on UI.
`} catch (PersistenceException e) {
            Log.error("Error on Persist data: ", e);
            return false;
        } catch (Exception e) {
            Log.error("Internal Server Error : ", e);
            throw new Exception(e);
        }`
You will also change Account Controller also accordingly. may be you can return Response instead of boolean.
`try {
            return Response.status(200).entity(accountService.insert(account)).build();
        } catch (Exception e) {
            return Response.status(500).entity(e.getMessage()).build();
        }`